### ISSUE TITLE
Evaluate Snowpack as an alternative to Rollup and Webpack

### ISSUE THREAD

**Issue by geigerzaehler** [radicle-dev/radicle-upstream#1704]
We have been having issus with rollup (https://github.com/radicle-dev/radicle-upstream/issues/884) especially with it’s slow bundling times (https://github.com/radicle-dev/radicle-upstream/issues/1658). We are considering Webpack as an alternative. It is a mature project that has proven itself in production.

[Snowpack](http://snowpack.dev/) is another alternative, promising fast development combined with bundling for production using Webpack. Snowpacks approach (and what makes it fast) is serving transpield ES modules separately from an HTTP server. Unfortunately, this is would require a change in our setup since now Electron loads the UI bundle from disk. It would be possible to have Electron load the files from a web server in development and from a bundle on disk in production. But this would result in diverging setups which could be the source of problems in production that are not caught in development or testing and would create a higher maintenance burden.

On the other hand, in my tests, Webpack rebuild the development bundle in under a second in watch mode. This eliminates the advantage that Snowpack has.

In summary, I believe that we should move forward with Webpack.

**Comment by xla** [comment]
Are adoption and preference in the ecosystem a consideration here? Specifically will we see the Svelte community lean towards one or the other and therefore more likely support and advance the tooling on top of either option?

**Comment by xla** [comment]
https://svelte.dev/blog/whats-the-deal-with-sveltekit

**Comment by geigerzaehler** [comment]
> Are adoption and preference in the ecosystem a consideration here?Specifically will we see the Svelte community lean towards one or the other and therefore more likely support and advance the tooling on top of either option?

Adoption and preference are definitely a consideration. At the moment Rollup seems to be the most widely used bundler in the Svelte community and (as mentioned in the post you linked) Snowpack seems to be gaining momentum which might result in better integration. But we also have to keep in mind that adopting Rollup lead to a lot of maintenance effort and now we need to drop it. If there is a significant advantage in using Snowpack over Webpack I’d be happy to put in the work myself to switch to it. But for now I believe we benefit from something that is mature and proven.

**Comment by rudolfs** [comment]
Do we still want to do this or are we happy with rollup for now @geigerzaehler?

**Comment by geigerzaehler** [comment]
We can close this. As mentioned in the description, we decided to go with Webpack and there’s no need for us to switch since it’s working just fine.
